adapter/internal/ga/client.go

```diff
@@ -288,13 +290,98 @@ func addAPIToChannel(resp *discovery.DiscoveryResponse) {
 			IsDeployEvent:    false,
 			OrganizationUUID: gaAPI.OrganizationUUID,
 			RevisionUUID:     gaAPI.RevisionUUID,
+			DeployedEnv:      gaAPI.DeployedEnv,
 		}
 		GAAPIChannel <- event
 		delete(apiRevisionMap, apiEntry)
 		logger.LoggerGA.Infof("API Undeploy event is added to the channel. : %s", apiEntry)
 	}
 }
 
+func addAPIWithEnvToChannel(resp *discovery.DiscoveryResponse) {
+
+	// removedAPIEnvMap is used To keep track of the APIs needs to be deleted.
+	var removedAPIEnvMap = make(map[string]map[string]*ga_model.Api)
+
+	if !isFirstResponse {
+		for k, v := range apiEnvRevisionMap {
+			cp := make(map[string]*ga_model.Api)
+			for l, w := range v {
+				cp[l] = w
+			}
+			removedAPIEnvMap[k] = cp
+		}
+	}
+
+	var startupAPIEventArray []*APIEvent
+	// Even if there are no resources available within ga, an empty but non-nil array would be set
+	startupAPIEventArray = make([]*APIEvent, 0)
+	for _, res := range resp.Resources {
+		api := &ga_model.Api{}
+		err := ptypes.UnmarshalAny(res, api)
+
+		if err != nil {
+			logger.LoggerGA.Errorf("Error while unmarshalling: %s\n", err.Error())
+			continue
+		}
+
+		currentGAAPI, apiFound := apiEnvRevisionMap[api.ApiUUID][api.DeployedEnv]
+		if apiFound {
+			if currentGAAPI.RevisionUUID == api.RevisionUUID {
+				logger.LoggerGA.Debugf("Current GA API revision ID and API event revision ID: %s in "+
+					"environment: %s are equal", currentGAAPI.RevisionUUID, currentGAAPI.DeployedEnv)
+				delete(removedAPIEnvMap[api.ApiUUID], api.DeployedEnv)
```

Comment:
Possible NPE if `removedAPIEnvMap` does not contain api.APIUUID.

Reply:
In a perfect scenario, this won't be null but due to bug let's say per API, per Env there would be two deployments and isFirstResponse is true this could occur. 
Let's make sure code does not break avoiding the NPE.

Reply:
add a check prior to deleting https://github.com/wso2/product-microgateway/commit/ae2c06882353bda7a59101381ef3e82f88667a21